stackslib/src/chainstate/stacks/boot/signers.clar

```diff
@@ -13,6 +13,10 @@
 (define-read-only (stackerdb-get-signer-slots)
 	(ok (var-get stackerdb-signer-slots)))
 
+(define-read-only (get-signer-slots (signer principal) (reward-cycle uint))
+	(ok u1)
+)
```

Comment:
For now, the function `get-signer-slots` can look like this:

```clarity
(define-read-only stackerdb-get-signer-slots (signer-index uint)
	(ok (element-at (var-get stackerdb-signer-slots) signer-index))
)
```

Signers need to know their own index so it can be passed to the vote function. You can call it wrapped in the proper `at-block` from `pox-4-vote` to get the slot you want.

It is not the most efficient way but it will unblock the feature. We can later do a runtime cost assessment and decide to change it.

For example, it can also be an indexed map much like the PoX reward set info, but then the signer binary would have to do multiple RPC calls to fetch all the slots (instead of one).

If we don't mind duplicating the data then we can have both the list and the indexed map.

cc @jferrant

Reply:
Should this be part of a different PR?

Reply:
added in this PR